internal/reminder/metrics_server.go

```diff
@@ -0,0 +1,87 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package reminder
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/rs/zerolog"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+)
+
+const (
+	metricsPath       = "/metrics"
+	readHeaderTimeout = 2 * time.Second
+)
+
+func (r *reminder) startMetricServer(ctx context.Context, mpReady chan<- struct{}) error {
+	logger := zerolog.Ctx(ctx)
+
+	prometheusExporter, err := prometheus.New(
+		prometheus.WithNamespace("reminder"),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create Prometheus exporter: %w", err)
+	}
+
+	res := resource.NewWithAttributes(
+		semconv.SchemaURL,
+		semconv.ServiceName("reminder"),
+		// TODO: Make this auto-generated
+		semconv.ServiceVersion("v0.1.0"),
+	)
+
+	mp := sdkmetric.NewMeterProvider(
+		sdkmetric.WithReader(prometheusExporter),
+		sdkmetric.WithResource(res),
+	)
+
+	otel.SetMeterProvider(mp)
+
+	// Indicates that a global MeterProvider is available
+	close(mpReady)
+
+	mux := http.NewServeMux()
+	mux.Handle(metricsPath, promhttp.Handler())
```

Comment:
Why do you need an `http.ServeMux` here, rather than simply passing `promhttp.Handler` to the `http.Server` below?

Reply:
Do we want to serve the metrics on the root path? I don't see any problem with that, but it's just that I've seen `/metrics` being commonly used.

Reply:
(On the other hand, I don't have a strong complaint about putting this at `/metrics`)